File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        
        for ( int i = 0; i < nums.length; i++){
            int reqNum = target - nums[i];
            if (numMap.containsKey(reqNum)){
                return new int[]{i, numMap.get(reqNum)};
            }
            numMap.put(nums[i], i);
        }
        return new int[]{};
        
        
    }
}

File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dict_nums = {}
        for i,v in enumerate(nums):
            dict_nums[v] = i
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in dict_nums and dict_nums[diff] != i:
                return [i, dict_nums[diff]]